src/Filters/remez_fir.jl

```diff
@@ -0,0 +1,829 @@
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+  c. Neither the name of Enthought nor the names of the SciPy Developers
+     may be used to endorse or promote products derived from this software
+     without specific prior written permission.
+
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS
+BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+
+--- end of scipy license
+
+The remez and associated functions were extracted from sigtoolsmodule.c,
+with the following header comment:
+
+SIGTOOLS module by Travis Oliphant
+
+Copyright 2005 Travis Oliphant
+Permission to use, copy, modify, and distribute this software without fee
+is granted under the SciPy License.
+
+C CODE BANNER
+
+/********************************************************
+ *
+ *  Code taken from remez.c by Erik Kvaleberg which was 
+ *    converted from an original FORTRAN by:
+ *
+ * AUTHORS: JAMES H. MCCLELLAN
+ *
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING AND COMPUTER SCIENCE
+ *         MASSACHUSETTS INSTITUTE OF TECHNOLOGY
+ *         CAMBRIDGE, MASS. 02139
+ *
+ *         THOMAS W. PARKS
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING
+ *         RICE UNIVERSITY
+ *         HOUSTON, TEXAS 77001
+ *
+ *         LAWRENCE R. RABINER
+ *         BELL LABORATORIES
+ *         MURRAY HILL, NEW JERSEY 07974
+ *
+ *  
+ *  Adaptation to C by 
+ *      egil kvaleberg
+ *      husebybakken 14a
+ *      0379 oslo, norway
+ *  Email:
+ *      egil@kvaleberg.no
+ *  Web:
+ *      http://www.kvaleberg.com/
+ *
+ *********************************************************/
+
+=============================================#
+
+#using Formatting
+
+# RemezFilterType:
+#    Type I and II symmetric linear phase: neg==0   (filter_type==bandpass)
+#    Type III and IV negative symmetric linear phase: neg==1   (filter_type==hilbert or differentiator)
+@enum RemezFilterType filter_type_bandpass=1 filter_type_differentiator=2 filter_type_hilbert=3
+
+
+
+"""/*
+ *-----------------------------------------------------------------------
+ * FUNCTION: lagrange_interp (d)
+ *  FUNCTION TO CALCULATE THE LAGRANGE INTERPOLATION
+ *  COEFFICIENTS FOR USE IN THE FUNCTION gee.
```

Comment:
While this docstring may be of historical interest (?), it's pretty pointless in its current form. It doesn't explain the parameters, and it refers to a non-existing function `gee`. Also, the embedded C-style comment marks don't make too much sense when interpreted as markdown. Not sure what to do here though. Come up with a proper docstring? Convert into a Juila style comment?